benchmark/v4.0-performance-benchmarking-with-tpch.md

```diff
@@ -0,0 +1,200 @@
+| TiKV      | 3.0 and 4.0   |
+| TiFlash   | 4.0        |
+| tiup-bench | 0.2      |
+
+### Parameter configuration
+
+#### v3.0
+
+For v3.0, TiDB, TiKV and PD use the default parameter configuration.
+
+##### Global variable configuration
+
+{{< copyable "sql" >}}
+
+```sql
+set global tidb_distsql_scan_concurrency = 30;
+set global tidb_projection_concurrency = 16;
+set global tidb_hashagg_partial_concurrency = 16;
+set global tidb_hashagg_final_concurrency = 16;
+set global tidb_hash_join_concurrency = 16;
+set global tidb_index_lookup_concurrency = 16;
+set global tidb_index_lookup_join_concurrency = 16;
+```
+
+#### v4.0
+
+For v4.0, TiDB uses the default parameter configuration.
+
+##### TiKV configuration
+
+{{< copyable "" >}}
+
+```yaml
+readpool.storage.use-unified-pool: false
+readpool.coprocessor.use-unified-pool: true
+```
+
+##### PD configuration
+
+{{< copyable "" >}}
+
+```yaml
+replication.enable-placement-rules: true
+```
+
+##### TiFlash configuration
+
+{{< copyable "" >}}
+
+```yaml
+logger.level: "info"
+learner_config.log-level: "info"
+```
+
+##### Global variable configuration
+
+{{< copyable "sql" >}}
+
+```sql
+set global tidb_allow_batch_cop = 1;
+set global tidb_opt_distinct_agg_push_down = 1;
+set global tidb_distsql_scan_concurrency = 30;
+set global tidb_projection_concurrency = 16;
+set global tidb_hashagg_partial_concurrency = 16;
+set global tidb_hashagg_final_concurrency = 16;
+set global tidb_hash_join_concurrency = 16;
+set global tidb_index_lookup_concurrency = 16;
+set global tidb_index_lookup_join_concurrency = 16;
+```
+
+### Test plan
+
+#### Hardware prerequisite
+
+To avoid TiKV and TiFlash racing for disk and I/O resources, mount the two NVMe SSD disks configured on EC2 as `/data1` and `/data2`. Deploy TiKV on `/data1` and deploy TiFlash on `/data2`.
+
+#### Test process
+
+1. Deploy TiDB v4.0 and v3.0 using TiUP.
+
+2. Import the TPC-H 10 data using the bench tool of TiUP.
+
+    * Execute the following command to import data into v3.0:
+
+        {{< copyable "bash" >}}
+
+        ```bash
+        tiup bench tpch prepare \
+        --host ${tidb_v3_host} --port ${tidb_v3_port} --db tpch_10 \
+        --sf 10 \
+        --analyze --tidb_build_stats_concurrency 8 --tidb_distsql_scan_concurrency 30
+        ```
+
+    * Execute the following command to import data into v4.0:
+
+        {{< copyable "bash" >}}
+
+        ```bash
+        tiup bench tpch prepare \
+          --host ${tidb_v4_host} --port ${tidb_v4_port} --db tpch_10 --password ${password} \
+          --sf 10 \
+          --tiflash \
+          --analyze --tidb_build_stats_concurrency 8 --tidb_distsql_scan_concurrency 30
+        ```
+
+3. Execute the TPC-H query.
+
+    1. Download the TPC-H SQL query file:
+
+        {{< copyable "" >}}
+
+        ```bash
+        git clone https://github.com/pingcap/tidb-bench.git && cd tpch/queries
+        ```
+
+    2. Execute the query and record the executing time of the query.
```

Comment:
Should be "Execute those queries record the executing time of each query"?

Reply:
Adopted, thanks.